### ISSUE TITLE
HTML popup when players spawn

### ISSUE THREAD

**Issue by wrefgtzweve** [DrVrej/VJ-Base#77]
Players are getting this html panel when they join the server, the server is using VJ base from the workshop.
Is there any way to disable this popup? I've tried looking where it's coming from and why it's 404 but i couldn't find anything.

![image](https://user-images.githubusercontent.com/69946827/177684399-2d483637-5705-4eb0-b258-6b80032b80fb.png)

**Comment by DrVrej** [comment]
This appears when an addon is using the old checking system and VJ Base isn't present. I don't run servers, but I assume your users should also have VJ Base installed.

**Comment by wrefgtzweve** [comment]
https://steamcommunity.com/sharedfiles/filedetails/?id=152529683 still uses said old checking system, even when VJBase is present it will display this

**Comment by DrVrej** [comment]
You're reading the code completely wrong, you skipped all the code above it. For your own sake since this is the second time you interpreted code wrong, I highly recommend learning basics of Lua and programming before attempting to understand or edit any code. There many great tutorials online, check them out. 👍

**Comment by wrefgtzweve** [comment]
VJ base is installed and the addon works fine.
This is failing for some reason.
![image](https://user-images.githubusercontent.com/69946827/178066734-c0ed5ced-19b2-4fee-8cf7-eddd2fe6c928.png)

Which in term gives every player that joins the server this faulty popup.
![image](https://user-images.githubusercontent.com/69946827/178066770-8e43ae48-2572-4ca0-ae11-3f739a62e036.png)

Simply discarding an issue where every player gets a useless html panel in their face each time they join a server as "not knowing lua" is not only just being a dick but also showing that you clearly don't even understand your very own code.
The server is uses both workshop addons, unless you somehow managed to mess up the workshop addons to such a point where your own addons fail to properly detect each other. The NPC's work fine, yet people still get these popups.

![image](https://user-images.githubusercontent.com/69946827/178067430-0d329e85-2e2c-424a-87cc-fa4bc1e34ea2.png)
The check passes fine on server.

Yes i misread what VJF was doing there as i did not expect you to make a global for a panel.

**Comment by DrVrej** [comment]
The fact that you still don't understand the reason for `VJF` and think that I messed up the file detection shows that you lack understanding of basic programming and I am not saying there is anything wrong with that. I was genuinely trying to help you for your own good because I know other developers will not respond nicely when someone misreads basic code.

As for the file detection function, if it is failing, what do you want me to do? It's a built-in Garry's Mod function. Another user found an issue with the function too, you can view it here: https://wiki.facepunch.com/gmod/file.Exists But by the time anyone realized this issue, there was already 10000s of addons using it, including 100s of VJ addons. Basically in some dedicated servers, the function will simply fail and return false. The workaround for the issue is performance heavy and in many cases, not a viable solution at all. The Garry's Mod developers themselves need to fix this function, which will solve all of these mysterious "missing file" issues in addons.

**Comment by wrefgtzweve** [comment]
So the issue exists and you aren't able to fix it due to incompetence, thanks you could've mentioned that earlier. Then I'm done here.

For anyone that reads this thread on how to fix this simple do this:

Run this on autorun client to prevent the popup:
![image](https://user-images.githubusercontent.com/69946827/178071222-9cf0a6d6-0de5-47f2-a64b-51de955d52dc.png)

Run this on autorun serverside if you don't want the annoying text message on spawn and want to remove his narcissistic net message.
https://github.com/DrVrej/VJ-Base/blob/83c5e48ef3f0aff1ede7716a23c087a6a11f3eb3/lua/autorun/vj_base_autorun.lua#L60-L75
![image](https://user-images.githubusercontent.com/69946827/178071262-0e44503a-c2a2-46fb-aa72-6f7a8900f53b.png)

**Comment by DrVrej** [comment]
Issue was found in 2020 and was never announced to developers. It's not the fault of addon creators, in fact I only found out about the issue today when I looked up the function, so not sure how you want me to have mentioned it earlier. Also the only narcissistic person here is you, I gave you a friendly genuine suggestion to learn basic Lua and you turned into a egotistical brat. People like you are the reason why there are so many issues in the modern world. You refuse to accept that you lack knowledge or skill of something. You embrace those who say falsely tell you "Its correct" while knowing you are wrong, and hate on those who try to actually help by suggesting you to expend your knowledge in something.

**Comment by wrefgtzweve** [comment]
Yeah sure man, if that's what keeps you sane sure. I just don't want to see my players being bothered by some guys bad code / decisions. I call you a narcissist because apparently you feel the need to add 3 different ways to display your name in the chat when you join a server, if that isn't narcissism i don't know what is. 

![image](https://user-images.githubusercontent.com/69946827/178075068-8c173025-68c3-470d-9e78-7f1212b080a3.png)
![image](https://user-images.githubusercontent.com/69946827/178075074-efa93c5f-6442-4db9-a2da-256cd582bc09.png)
![image](https://user-images.githubusercontent.com/69946827/178075080-890166c5-e576-49fb-a2bd-326674165201.png)
![image](https://user-images.githubusercontent.com/69946827/178075131-5bec46d6-9866-4c66-b3d3-66d0ac3fbe61.png)


And "I gave you a friendly genuine suggestion to learn basic Lua and you turned into a egotistical brat", come on lmao.

**Comment by DrVrej** [comment]
Shitting an old addon from 2012 but can't even understand a simply if-then statement 🤣
Also that's 2 not 3, but I'll make sure to include it 10 times in the next update, just for you 😘

**Comment by Cpt-Hazama** [comment]
Man, it just so happens after all these years that a random server owner has cracked the code of the mysterious VJ Base pop up 👏 congratulations sir! May all 5 players on your server who didn't install VJ Base be saved from the old mod pop up 🙏

**Comment by IvanVladimir04** [comment]
> Yeah sure man, if that's what keeps you sane sure. I just don't want to see my players being bothered by some guys bad code / decisions. I call you a narcissist because apparently you feel the need to add 3 different ways to display your name in the chat when you join a server, if that isn't narcissism i don't know what is.
> 
> ![image](https://user-images.githubusercontent.com/69946827/178075068-8c173025-68c3-470d-9e78-7f1212b080a3.png) ![image](https://user-images.githubusercontent.com/69946827/178075074-efa93c5f-6442-4db9-a2da-256cd582bc09.png) ![image](https://user-images.githubusercontent.com/69946827/178075080-890166c5-e576-49fb-a2bd-326674165201.png) ![image](https://user-images.githubusercontent.com/69946827/178075131-5bec46d6-9866-4c66-b3d3-66d0ac3fbe61.png)
> 
> And "I gave you a friendly genuine suggestion to learn basic Lua and you turned into a egotistical brat", come on lmao.

Way to show your ego and lack of intelligence.

**Comment by TheReddestBread** [comment]
I just hate Armenians

I will elaborate no further.

**Comment by IvanVladimir04** [comment]
"ArMeNiAn SpOtTeD" 🤓🤓🤓🤓

Go do something more productive

**Comment by TheReddestBread** [comment]
How about you climb out of your mudhole shit country and do something productive with your life rather than code shitty ass addons and be a egotistical ass when someone criticizes ur work.

**Comment by TheReddestBread** [comment]
"🤓🤓🤓🤓"

**Comment by IvanVladimir04** [comment]
> How about you climb out of your mudhole shit country and do something productive with your life rather than code shitty ass addons and be a egotistical ass when someone criticizes ur work.

Are you feeling better now? cried enough or do you want me to get your baby milk?

**Comment by Cpt-Hazama** [comment]
https://user-images.githubusercontent.com/7193583/178077564-d9a5b0e0-1322-4883-81f8-23882198e5d1.mp4



> "🤓🤓🤓🤓"

**Comment by wrefgtzweve** [comment]
> Shitting an old addon from 2012 but can't even understand a simply if-then statement 🤣 Also that's 2 not 3, but I'll make sure to include it 10 times in the next update, just for you 😘

So this bug is in the code since 2012 and you never bothered fixing it? 💀

3* console command, button in the menu and a message on join. Unless you forgot what you put in your own addon, seems like a pattern at this point.

**Comment by TheReddestBread** [comment]
https://user-images.githubusercontent.com/80224326/178077684-508dfda3-49a0-45d9-bcd0-cc4b1975fcfb.mp4

**Comment by DrVrej** [comment]
> I just hate Armenians
> 
> I will elaborate no further.

Cope harder 🇦🇲🇦🇲🇦🇲🇦🇲🇦�
![image](https://user-images.githubusercontent.com/6934542/178077778-ac716c8c-d4c6-4a95-8397-5e4ca6df6c4f.png)
�

**Comment by Cpt-Hazama** [comment]
> 

So he needs to go to facepunch studios, hack into their computers, and fix a Garry's Mod bug himself? 🤡

**Comment by DrVrej** [comment]
> > Shitting an old addon from 2012 but can't even understand a simply if-then statement 🤣 Also that's 2 not 3, but I'll make sure to include it 10 times in the next update, just for you 😘
> 
> So this bug is in the code since 2012 and you never bothered fixing it? 💀
> 
> 3* console command, button in the menu and a message on join. Unless you forgot what you put in your own addon, seems like a pattern at this point.

Let me teleport to one of the developer's house and force them to fix it, got it 🤣
Also I love how you sent your friend / alt to post racist comments and videos, keep showing your true colors 😀

**Comment by TheReddestBread** [comment]
![unkghjkghjkghjknown](https://user-images.githubusercontent.com/80224326/178077977-231e7b04-30a9-4453-b448-6ae96c60d668.png)
your country and "people" mean nothing to the civilized world

**Comment by TheReddestBread** [comment]
![unkghjkghjkghjknown](https://user-images.githubusercontent.com/80224326/178078021-bae0cf82-f20f-4c26-8030-484b43fe411d.png)

**Comment by TheReddestBread** [comment]
![jhklhjlhjllkjkhllhjl](https://user-images.githubusercontent.com/80224326/178078061-533b6709-6fda-4bd9-a6fc-1c25f4798fc8.PNG)

**Comment by TheReddestBread** [comment]
![unknsgggggown](https://user-images.githubusercontent.com/80224326/178078136-a4c171ab-40be-452a-a897-31c58da6db28.png)

**Comment by wrefgtzweve** [comment]
> > > Shitting an old addon from 2012 but can't even understand a simply if-then statement 🤣 Also that's 2 not 3, but I'll make sure to include it 10 times in the next update, just for you 😘
> > 
> > 
> > So this bug is in the code since 2012 and you never bothered fixing it? 💀
> > 3* console command, button in the menu and a message on join. Unless you forgot what you put in your own addon, seems like a pattern at this point.
> 
> Let me teleport to one of the developer's house and force them to fix it, got it 🤣 Also I love how you sent your friend / alt to post racist comments and videos, keep showing your true colors 😀

i posted it in a public discord to show your shitty responses, i don't even know the guy nor support his actions.

**Comment by IvanVladimir04** [comment]
> > > > Shitting an old addon from 2012 but can't even understand a simply if-then statement 🤣 Also that's 2 not 3, but I'll make sure to include it 10 times in the next update, just for you 😘
> > > 
> > > 
> > > So this bug is in the code since 2012 and you never bothered fixing it? 💀
> > > 3* console command, button in the menu and a message on join. Unless you forgot what you put in your own addon, seems like a pattern at this point.
> > 
> > 
> > Let me teleport to one of the developer's house and force them to fix it, got it 🤣 Also I love how you sent your friend / alt to post racist comments and videos, keep showing your true colors 😀
> 
> i posted it in a public discord to show your shitty responses, i don't even know the guy nor support his actions.

You called in a racist turk who wants to genocide armenia, what a great server owner you are

**Comment by IvanVladimir04** [comment]
> ![unknsgggggown](https://user-images.githubusercontent.com/80224326/178078136-a4c171ab-40be-452a-a897-31c58da6db28.png)

Still mad armenia will exist forever?

**Comment by IMPULSEPLANTER6** [comment]
SHUT UP NIGGER

**Comment by IMPULSEPLANTER6** [comment]
kill yourself

**Comment by IMPULSEPLANTER6** [comment]
ASAP

**Comment by wrefgtzweve** [comment]
Yeah imma leave i don't support any of this.

**Comment by IMPULSEPLANTER6** [comment]
THIS ADDON IS SO ASS

**Comment by IMPULSEPLANTER6** [comment]
https://cdn.discordapp.com/attachments/764150590812192798/995095860025233468/evil.mp4

**Comment by TheReddestBread** [comment]
https://user-images.githubusercontent.com/80224326/178078518-d5493402-8dd8-4c0b-94af-6bad8ecab73c.mp4

**Comment by IMPULSEPLANTER6** [comment]
im such a girlboss

**Comment by Cpt-Hazama** [comment]
https://user-images.githubusercontent.com/7193583/178078704-3c698ae9-a911-4044-989d-d590f1e5d89a.mp4

**Comment by DrVrej** [comment]
> Yeah imma leave i don't support any of this.

Instead of putting so much energy to send people to attack modders, you should put that energy to letting the GMod developers know about the bug and explain to them how essential it is to fix it. Also don't bother deleting your comments, GitHub saves everything unlike other platforms:

![image](https://user-images.githubusercontent.com/6934542/178079229-dc3b3046-d9eb-49f7-9cbd-2a2e3bfba765.png)

**Comment by wrefgtzweve** [comment]
Yeah i don't want to be associated with said people. I just showed what i was doing and people had to join and fuck it up.  Sorry for that part, i didn't expect it would go like this.

**Comment by IvanVladimir04** [comment]
That's not gonna convince anyone, it's obvious this was planned, more than one racist user joined in the instant you were being replied to

**Comment by wrefgtzweve** [comment]
Yeah man, i totally spend my time making evil plans to raid github issues!!!!

**Comment by IvanVladimir04** [comment]
Because you have nothing better to do with your time it seems

**Comment by wrefgtzweve** [comment]
Obviously!

**Comment by ZombieWars** [comment]
> Obviously!

thanks for the help, your notes helped me!!!
